### PR TITLE
Hydra Agent GUI Console Implementation

### PR DESCRIPTION
**Pull request by Guttz**
Related to: #124

Porting commits/progress from https://github.com/HTTP-APIs/hydra-python-agent/pull/125

This PR will implement a user interface for the Python Agent.

### CODE REVIEW

**Comment by Guttz** [comment]
Today I changed my approach on how the ApiDoc was being used according to a suggestion made by Vishal. Now I'm using the built-in generate() function from doc_maker instead of serializing the whole object. This took some time but now all endpoints are fetched from EntryPoint, and all are shown to the user: both collection and non-collection.

Also, I and Vishal discussed in more detail the two missing endpoints `send_command`/`hydra_doc` and settled some pending points. The first, Vishal already implemented the first version and I'll test soon. The second is being developed. 

Now the Operations component was developed and automatically shows the supported operations. The Properties component is being a bit more challenging and I still have some work left on it. Small gif:

console2

**Comment by Guttz** [comment]
Now finally the properties component is being dynamically built according to the selected endpoint. I've been having some problems with processing the Hydra Doc in the most adequate way(Vishal is helping me) and maintaining the state.

 The whole component is partially complete, I'll finish it up passing the parameters to the command/other small details tomorrow and complement some decisions I had to make when processing the Hydra Doc.

 Small progress gif: 
console3

**Comment by Guttz** [comment]
I've set the parameters and started to integrate with the send-command backend done by Vishal. Which means making the HTTP request to the backend and showing the result properly to the user. Now I'm fixing some decoding problems but will have a full first working version until Sunday. 

console4

**Comment by Guttz** [comment]
The first working version is done. The user can query through the endpoints and make the enabled requests by the Python Agent while seeing the command produced. By running a command the output is shown below. The user can also switch the hydrus server by inserting a URL for another server.

https://www.youtube.com/watch?v=IAgThJGfQ08&feature=youtu.be

There were some decision making I had to make regarding how to process Hydra Doc. The main one was on how to enable the user to make requests for the Resources and The Collections without simply creating 15+ "endpoints". I've added the Resource ID field and the Agent itself depending on how the user is interacting with the properties decides if it will go for the collection or resource. There are still other minor bugs that I'd like to fix to make the UX better/responsiveness. I'll open the issues and list for these first. As we've settled in our call I'll start listing the documentation to be updated and finish off details according to time available.

**Comment by Guttz** [comment]
There were some decision making I had to make regarding how to process Hydra Doc. The main one was on how to enable the user to make requests for the Resources and The Collections without simply creating 15+ "endpoints". I've added the Resource ID field and the Agent itself depending on how the user is interacting with the properties decides if it will go for the collection or resource. There are still other minor bugs that I'd like to fix to make the UX better and also UI/responsiveness enhancements.

 I'll open the issues and list the things I find important about these. As we've settled in our call I'll start listing the documentation to be updated in the whole org(including to run this GUI) and finish off details according to the time available.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+flask
+Flask-Cors
+-e git+https://github.com/RedisGraph/redisgraph-py#egg=redisgraph
+-e git+https://github.com/HTTP-APIs/hydra-python-core#egg=hydra_python_core
+-e git+https://github.com/HTTP-APIs/hydra-python-agent@develop#egg=hydra_agent
```
Comment (xadahiya) [inline requirements.txt]: I've created a PR on the agent to sync develop with the master. @Guttz please review that, then change the branch here to master and then we can merge.

**Comment by Mec-iS** [comment]
Great people!

File: requirements.txt (hunk 1 above)
Comment (Guttz) [reply requirements.txt]: Okay! I'll work on the documentation and double-check everything the next days so we merge both of them with proper docs.

**Pull request merged**